### PR TITLE
[Driver][SYCL] Fix crash with empty -fsycl-targets option

### PR DESCRIPTION
**Pull request by mdtoguchi**
When -fsycl-targets= -fsycl-targets=arg is passed on the command line, the empty -fsycl-targets= is processed and crashes due to the empty argument.  Update the specific option parsing to use 'getLastArg' to avoid this situation.  When the last arg has an empty argument, the driver will diagnose this situation and emit a friendly diagnostic.

### CODE REVIEW

**Comment by srividya-sundaram** [comment]
> When the last arg has an empty argument, the driver will diagnose this situation and emit a friendly diagnostic.

I'm not able to see the corresponding code changes for this in the PR.

**Comment by mdtoguchi** [comment]
> > When the last arg has an empty argument, the driver will diagnose this situation and emit a friendly diagnostic.
> 
> I'm not able to see the corresponding code changes for this in the PR.

There are no explicit changes that reflect this - it is existing behavior:
`> clang++ -fsycl -fsycl-targets= ~/a.cpp
clang++: warning: joined argument expects additional value: '-fsycl-targets=' [-Wunused-command-line-argument]
`

File: clang/test/Driver/sycl-offload.cpp [hunk 1]
```diff
@@ -104,3 +104,7 @@
 // CHECK_FSYCL_FP64_CONV_EMU_WIN-NOT: clang{{.*}} "-cc1" "-triple x86_64-unknown-linux-gnu" {{.*}} "-fsycl-fp64-conv-emu"
 // CHECK_FSYCL_FP64_CONV_EMU_WIN-DAG: clang{{.*}} "-cc1" "-triple" "spir64_gen{{.*}}" "-fsycl-fp64-conv-emu"
 // CHECK_FSYCL_FP64_CONV_EMU_WIN-DAG: clang-offload-packager{{.*}} "--image=file={{.*}}.bc,triple=spir64_gen-unknown-unknown,arch=,kind=sycl,compile-opts={{.*}}-options -ze-fp64-gen-conv-emu{{.*}}"
+
+/// Compilation checks to make sure an early empty -fsycl-targets does not
+/// crash.
+// RUN: %clangxx -### -fsycl -fsycl-targets= -fsycl-targets=spir64 %s
```
Comment (srividya-sundaram) [inline clang/test/Driver/sycl-offload.cpp:110]: Is this supposed to be just a RUN command?
If yes, can we instead check if `spir64` target flow is achieved?

Comment (mdtoguchi) [reply clang/test/Driver/sycl-offload.cpp:110]: It is just a check to see if the command returns an error.  I will update to check for the device compilation.

**Comment by mdtoguchi** [comment]
@intel/llvm-gatekeepers, this change looks ready to merge, please take a look.  Thanks!

**Pull request merged**